ckanext/dhis2harvester/templates/source/dhis2_credentials.html

```diff
@@ -8,18 +9,44 @@ <H2>
 </H2>
 {% endblock %}
 {% block connection_details %}
+<div class="advanced-config toggle-box dhis2-api">
 {% call form.input('dhis2_url', id='field-dhis2-url', label=_('DHIS2 URL'), value=data.dhis2_url, placeholder="e.g. https://play.dhis2.org/demo/", error=errors.dhis2_url, classes=['control-full'], is_required=True) %}
     <span class="info-block">
       {{ _('This should include the https:// part of the URL') }}
+      <div class="header">
+        <a href="#" class="toggle-btn toggled" data-module="toggle_box">{{_('Click here to optionally specify the DHIS2 API version')}}</a>
+      </div>
     </span>
 {% endcall %}
-{% call form.input('dhis2_api_version', id='field-dhis2-api-version', label=_('DHIS2 API version'), value=data.dhis2_api_version, placeholder="26", error=errors.dhis2_api_version, classes=['control-full']) %}
-  <span class="info-block">
-    {{ _('DHIS2 API version to be used. If empty the default version 26 will be used.') }}
-  </span>
-{% endcall %}
+
+  <div class="content"  style="display:none">
+    {% call form.input('dhis2_api_version', id='field-dhis2-api-version', label=_('DHIS2 API version'), value=data.dhis2_api_version, placeholder="26", error=errors.dhis2_api_version, classes=['control-full']) %}
+      <span class="info-block">
+        {{ _('DHIS2 API version to be used. If empty the default version 26 will be used.') }}
+      </span>
+    {% endcall %}
+  </div>
+</div>
+
 {% if data.dhis2_auth_token %}
-  {{ form.info(text=_('Auth token saved')) }}
+  {{form.info(text=_('A previous DHIS2 session is still active.  No need to log in again. '))}}
+  <div class="advanced-config toggle-box dhis2-login">
+  <div class="header">
+  <a href="#" class="toggle-btn toggled" data-module="toggle_box">{{_('Click here to change DHIS2 User')}}</a>
```

Comment:
Could you make clicking on the toggle link clear content of `dhis2_auth_token`? If the token is present it's processed with higher priority in the blueprint.
Other option is that we'd need to pass "update credentials" flag to the blueprint which sounds more complicated IMO.

Reply:
So you're just suggesting {% set dhis2_auth_token = "" %}?

Reply:
@tomeksabala Will change the order of precedence in the back end to make this work. I'll leve the PR open until this is done.